lms/services/exceptions.py

```diff
@@ -20,6 +20,10 @@ class ExternalRequestError(Exception):
     :arg message: A short error message for displaying to the user
     :type message: str
 
+    :arg error_code:
+        A code that identifies the error. Used by the frontend to display richer
+        error messages.
```

Comment:
We have some existing sub-classes of `ExternalRequestError` whose main function is to set an `error_code`. To me it makes sense that setting an error code, which is essentially an identifier for a richer version of `message`, should be built-in functionality.